### PR TITLE
cleanup: remove unused python imports

### PR DESCRIPTION
**Pull request by LanceEa**
## Description

After reviewing some of the dependabot PR's I noticed that some of them look as though they were no longer being used. So, this is to help get a better picture of what deps are actually being used and see which ones are no longer needed. This will help with dependabot and scanning tools so that we are not being flag for outdated deps that no longer get used.

The following was done:

- Executed `autoflake --remove-all-unused-imports  -r -i ./python` to
remove unused exports within the python.
- Reviewed changes to ensure re-exports were not removed.
- Executed `black ./python/` to fix lint issues
- Executed `pip-compile --allow-unsafe --no-build-isolation -q` to
update the requirements.txt to remove unused deps.
- Executed `isort ./python/` to ensure imports are sorted correctly
after updating them
- Executed `make generate` to ensure we capture updated dependenceis

## Related Issues
Example dependabot PR that is no longer used and we can remove the dependency. 

https://github.com/emissary-ingress/emissary/pull/4638

## Testing
CI Testing

## Checklist
- [x] **Does my change need to be backported to a previous release?**
 Yes, lets backport to `release/v2.5` to make it cleaner for cherry-picks. In general, may be cleaner to just run it there too instead because of the additional Envoy V2 files and some of these imports might still be used in the release/v2.5 branch.
- [ ] **I made sure to update `CHANGELOG.md`.**
- [x] **This is unlikely to impact how Ambassador performs at scale.**
- [x] **My change is adequately tested.**
- [ ] **I updated `DEVELOPING.md` with any any special dev tricks I had to use to work on this code efficiently.**
- [x] **The changes in this PR have been reviewed for security concerns and adherence to security best practices.**

### CODE REVIEW

**Comment by LanceEa** [comment]
Once PR https://github.com/emissary-ingress/emissary/pull/4642 is merged then the check branch will go away.

File: python/ambassador/cache.py [hunk 1]
```diff
@@ -242,7 +242,6 @@ def __init__(self, logger: logging.Logger) -> None:
         self.logger = logger
         self.logger.debug("NullCache: INIT")
         self.reset_stats()
-        pass
```
Comment (haq204) [inline python/ambassador/cache.py:245]: was this manually removed or from a linter? seems slightly odd for an import checker to remove. Maybe autoflake?

**Review by haq204 — COMMENTED** [review]
This makes me want to add pylint to our list of python linters which catches for things like unused imports, among other things.

I'm also fully relying on CI for testing

**Comment by LanceEa** [comment]
> This makes me want to add pylint to our list of python linters which catches for things like unused imports, among other things.
> 
> I'm also fully relying on CI for testing

Yeah, I think we can add something going forward. I wanted to use this to help cleanup some of the dependabot stuff because if I have time this week I will address some of the other opens.

Since I'm not a python person I don't have a preference on pylint vs autoflake vs black so we can discuss that as some follow-up work to do.

File: python/ambassador/cache.py (hunk 1 above)
Comment (haq204) [reply python/ambassador/cache.py:245]: yea `pass` is a no-op in python. it basically tells the python interpreter to do nothing

**Review by haq204 — DISMISSED** [review]
lgtm

**Comment by LanceEa** [comment]
Note: I will rebase and once we land the CI fix and that will also address the Branch version check too now that things have been tagged properly.

**Review by haq204 — APPROVED** [review]

**Review by AliceProxy — APPROVED** [review]
Love it, +1 to adding a linter that checks for this

**Pull request merged**